Meta Trader Base

Kurfursten opened the pull request:
I expressed four main issues with the Metaclub trader base in #17601

* Powergamers can simply count out how many tiles they need to go to get a roundstart salvage shuttle guaranteed
* It's the ultimate rapecave because there's no radio there ever
* An inconvenience to the traders, too, because no radio means they can't advertise
* Shuttles in transit can throw their flames onto Deep Space, sometimes incinerating traders (especially after the final shuttle leaves since it spits out so much for so long)

@Probe1 proposed a new, experimental traderbase which is more shuttle than base and can go to more than one destination. 

* Uses the new docking lights because it's so big to warn people when it's incoming
* Goes to transit space only when it's headed for the station
* Using the send shuttle button on the Trader PDA will send it to a random location of two you are NOT at
* Unlike other shuttles, can be deconstructed and requires its APC powered
* Somewhat vulnerable to monkey business / spooky aliums because of the space-exposed vent (you can also hitch a ride through space by hiding on that tile!) 
* Has two non-station stops, a solar farm and a casino

![untitled](https://user-images.githubusercontent.com/9782036/36928877-69d5f2e4-1e4f-11e8-8a1b-a33485d9a50e.png)


🆑 
* rscdel: Metaclub: The Vox have moved out of the deep space base. It remains for vault hunters, but has no Trader-related items, blueprinted areas, trader shuttle, or research ferry.
* rscadd: Metaclub: Vox station replaced with one large 'mothership' that needs to be powered and can stop at 3 locations (main station, solar farm, asteroid casino).
* rscadd: There is now a "Trader Shoal" joint account. All traders know the number and PIN. It starts with 0 credits and earns no wages over time. This allows Traders to use custom vendors.

spoonylove commented:
the fact that traders don't already have an account is silly when they deal with money so much.  I think a casino, with interesting prizes to be won could be fun for everyone. Custom Arcade machines that cost money with prizes other than toys, something like that.

MadmanMartian commented on maps/metaclub.dmm:
```diff
@@ -10433,81 +10433,81 @@
-"dTC" = (/obj/machinery/door_control{id_tag = "cryostasis"; name = "Cryostasis Chamber Control"; pixel_y = 30; range = 1},/obj/machinery/sleeper,/obj/machinery/light/small{dir = 4},/obj/effect/landmark/start{name = "Trader"},/turf/simulated/shuttle/floor/vox{icon_state = "floor3"},/area/mine/unexplored)
-"dTD" = (/obj/machinery/door/airlock/hatch{id_tag = "cryostasis"; name = "Cryostatis"},/turf/simulated/shuttle/floor/vox{icon_state = "floor3"},/area/mine/unexplored)
+"dTB" = (/turf/simulated/wall/r_wall,/area/shuttle/trade/start)
+"dTC" = (/obj/machinery/door/airlock/external,/obj/machinery/door/poddoor{density = 0; icon_state = "pdoor0"; id_tag = "vox_maint"; layer = 1.1; name = "Maintenance Blast Door"},/turf/simulated/floor/plating/vox,/area/shuttle/trade/start)
```
"layer" is forbidden.

Probe1 commented:
Nix the plasma glass.  Keep everything within the Vox aesthetic (dingy, second rate).  The tradeship itself is good.  I'd rather see it use sloped and angled turfs but we don't have those sprites to match rwalls so no big deal.   Add a smelting machine somewhere.

Kurfursten commented:
The problem with developing the "casino" too much is that it's on the roid so if it's just full of fun toys they'll immediately break in at shift start to steal the machines. The main reason I added an airlock to the outside was because otherwise I figured they'd just smash in.

But I could make it bigger and connect it to the top half with a bit of maintenance. Maybe even make it 'miners welcome' kind of thing. Depends on what people think of it.

spoonylove commented:
Spessmart can have mercbots so can a casino. Maybe do a ticket system like dave and busters, call it Dan and Bustas' win tickets cash em in nothing to steal if its like a biogenerator.

Durgen commented:
The casino definitely needs a poker table, decks of cards, a bar, etc

Kurfursten commented:
While I can make the casino larger, I don't know that I plan to do any intensive coding to add any special games as part of this PR.

I could look into adding some vending machines that are linked to pay to the trader account though so they can make money as people linger around.

Kurfursten commented:
![untitled](https://user-images.githubusercontent.com/9782036/36826600-1959bd2c-1cd4-11e8-82ef-6a7e93cd1980.png)

Casino version 2. Yes that is a miniature shooting range.

I was thinking about adding some vendors and making them linked to the upcoming trader account. I could even link the slot machines directly to the trader account instead of generating one with thousands of credits -- not sure if that would be good or bad. It would mean that the trader need to earn some money before they can really get people playing slots. Might just be best to leave them as normal slots.

Kurfursten commented:
Here's an in-game version I took testing. I also increased the firing range length, adjusted the default firing range target movement pattern to be more up-down oriented, swapped the lights for light tiles (and provided a light tile controller at the bar), and added a private room at the front (windows are one way glass, you can look out but not in)

![untitled](https://user-images.githubusercontent.com/9782036/36854626-10da086a-1d37-11e8-976b-4c7d75c30838.png)

Kurfursten commented:
I went ahead and changed most of the internal walls of the new trade ship to normal, non-reinforced walls.

Before I push this version here are there any final review requests? As to Probe's comment about the angled turfs, it might not be too jarring if I made the outside out of the salvage shuttle walls. I'm leaning toward doing that right now because I've noticed docking with the stations causes your ship walls to smooth with the dock walls.

Probe1 commented:
Well that's horrifying.

No the only thing that jumped out at me was the plasma windows not being thematic.

Durgen commented:
maybe some windows on the south side of the botany portion?

Kurfursten commented:
Done. I also swapped out the box cloner with a fishtank because I felt shrimp were more of a classic casino food, and changed the ship to use salvage ship parts (interior remains unreinforced walls).

I think this is good to go, this push also includes the small code update package.

Kurfursten commented:
OP updated with fresh render and info + 🆑

Kurfursten commented:
Please review or ready for merge

I will update the Trader page on the wiki once it's in

MadmanMartian commented on code/__HELPERS/globalaccess.dm:
```diff
@@ -3272,8 +3272,8 @@
-		if("voxresearch_shuttle")
-			global.voxresearch_shuttle=newval
+		//if("voxresearch_shuttle")
+		//	global.voxresearch_shuttle=newval
```
Just delete these rather than comment them out. File's big enough as it is.

DamianX replied:
>manually editing globalaccess

MadmanMartian replied:
> Can't get the python script to work if your byond is installed anywhere but the C: Drive

Fucking PATH

DamianX replied:
just `cd` into your byond\bin
smh

MadmanMartian commented on code/game/shuttles/voxresearch_shuttle.dm:
```diff
@@ -1,4 +1,4 @@
-var/global/datum/shuttle/voxresearch/voxresearch_shuttle = new(starting_area = /area/shuttle/voxresearch/station)
+/*var/global/datum/shuttle/voxresearch/voxresearch_shuttle = new(starting_area = /area/shuttle/voxresearch/station)
```
Just delete the file

Kurfursten commented:
Yes, those wooden areas you're describing are part of the abandoned plasmaman research base that is remaining in deep space. I did not unmap it, I just took away its Vox-related areas, so it's unpowered (I unmapped the APCs and instead put APC frames in their place), along with deleting the 2 shuttles and trade vendors.

The result is that you can still find it as a sort of unique space vault for meta only, but it's not attractive as an instant free bonus base for vaulthunting traders.

Kurfursten commented:
@MadmanMartian It is done

Kevak commented:
As a moderately regular trader. I'm concerned about how this will affect midround joining traders. As the 'shuttlebase' is mobile. And the vox arrival arc appears to target its 'initial dockpoint' primarily, and the casino location is on the literal other side of the asteroid (and where the shuttle will likely be docked most of the time if not on station); It appears as though it will be above and beyond the normal tier of difficulty for traders to get to their base unless I'm misparsing the arc. Traders lack any form of light or navigation for the roid. They already have a fair bit of difficulty navigating early on without needing to wander through the asteroid's tunnels aimlessly to get to the opposite side of the roid and no way to call for pickup and a 50/50 chance that they'll get shuttlecrushed if active traders are overzealous with the attempted pickup.

I'd suggest making the vox arrival arc target the casino rather than the 'ship' but considering it appears to by default be docked in that initial spot rather than 'inside' the casino, that might result in problems on rounds with no roundstart traders. I think it's a good idea but needs work.

TL;DR: Make it not hellish for latejoining traders to get to the vox base/shuttle.

Kevak commented:
Fair. A shuttle caller might be a good idea though as you noted.

Durgen commented:
I think this new ship needs a name, if only unofficially

Kurfursten commented:
I'm partial for Hindsight, because the control computer is on the same end as the propulsion

Kevak commented:
You forgot botany tools in the casino as a note.
